File: sorting/insertion_sort.py
from typing import List

def insertion_sort(nums: List[int]) -> List[int]:
    for i in range(1, len(nums)):  # Start from the second element
        key = nums[i]  # Current element to be inserted
        j = i - 1  # Index of the previous element

        # Move elements of nums[0..i-1], that are greater than key,
        # to one position ahead of their current position
        while j >= 0 and nums[j] > key:
            nums[j + 1] = nums[j]
            j -= 1
        
        # Insert the key into its correct position
        nums[j + 1] = key
    
    return nums

# Example usage:
nums = [12, 11, 13, 5, 6]
sorted_nums = insertion_sort(nums)
print("Sorted array:", sorted_nums)

File: old/sorting/insertion_sort.js
function insertionSort(nums) {
  for(let i=1;i<nums.length;i++) {
    let j=i;
    while(j>0 && nums[j-1]>nums[j]) {
      let tmp = nums[j-1];
      nums[j-1] = nums[j];
      nums[j] = tmp;
      j--;
    }
  }
  return nums;
}

console.log(insertionSort([3,1,5,8,2,6]));